src/stringProcessing/getLinkType.js

```diff
@@ -11,17 +11,20 @@ var urlHelper = require( "./url" );
  */
 
 module.exports = function( text, url ) {
-	var linkType = "other";
+	let linkType = "other";
 
-	var anchorUrl = urlHelper.getFromAnchorTag( text );
+	const anchorUrl = urlHelper.getFromAnchorTag( text );
+	const protocol = urlHelper.getProtocol( anchorUrl );
 
 	// Matches all links that start with http:// and https://, case insensitive and global
-	if ( anchorUrl.match( /https?:\/\//ig ) !== null ) {
+	if ( protocol === "http://" || protocol === "https://" ) {
 		linkType = "external";
 
 		if ( urlHelper.getHostname( anchorUrl ) === urlHelper.getHostname( url ) ) {
 			linkType = "internal";
 		}
+	} else if ( protocol === null && anchorUrl.charAt( 0 ) !== "#" ) {
+		linkType = "internal";
 	}
```

Comment:
worth nothing the logic of these conditions fails in case of implicit protocol links like `//www.yoast.com` which are identified as "other" while they actually are internal or external.